Gemfile

```diff
@@ -41,7 +41,7 @@ if RUBY_DESCRIPTION =~ /linux/
 end
 
 # `hanami assets` integration tests
-gem 'sass',          require: false
+gem 'sassc',         require: false
```

Comment:
Is this needed so that the gem is available globally (so `bundle install` doesn't need to be run in specs)?

Reply:
@landongrindheim TL;DR: We do one `bundle install` at the beginning of the test suite. It vendors all the needed gems for all the kind of tests.

---

We generate an gem index in the `vendor/cache` directory, which is the same used by Bundler to install the gems. At this point, the CI job is self-sufficient and it doesn't need to contact Rubygems anymore.

---

Why do we do this?

Because to make integration specs more realistic, for each test, we generate a brand new app with all the settings that it needs for that specific scenario under test (e.g. enabled sessions, use minitest, don't use `hanami-model`, use Subresource Integrity, etc..).

At this point, it would be onerous to contact Rubygems for each test, so we modify the default source of the `Gemfile` of each generated app:

```diff
- source "https://rubygems.org"
+ source "file:///path/to/vendor/cache"
```

Because of the gem index, `vendor/cache` can be used as a source in `Gemfile`
Then we run `bundle install --local`, which forces Bundler to use only the local gems, which are vendored in `vendor/cache`.

---

If you want a complete picture, read the code of `lib/hanami/devtools/integration/bundler.rb` from https://github.com/hanami/devtools.

Reply:
**Great** explanation! Thank you!

I know this has been merged now, but would it make sense to make `sassc` a run time dependency instead of [a development dependency of Assets](https://github.com/hanami/assets/blob/master/hanami-assets.gemspec#L34)? Would that remove the need to declare it in Hanami's Gemfile?

Reply:
Nope. That would turn `sassc` into the default compressor, but that isn't the case.